crates/bevy_window/src/raw_handle.rs

```diff
@@ -34,22 +31,27 @@ impl<W: 'static> Deref for WindowWrapper<W> {
     type Target = W;
 
     fn deref(&self) -> &Self::Target {
-        self.reference.downcast_ref::<W>().unwrap()
+        &self.reference
     }
 }
 
-/// A wrapper over [`RawWindowHandle`] and [`RawDisplayHandle`] that allows us to safely pass it across threads.
+trait WindowTrait: HasWindowHandle + HasDisplayHandle {}
+impl<T: HasWindowHandle + HasDisplayHandle> WindowTrait for T {}
+
+/// A wrapper over [`HasWindowHandle`] and [`HasDisplayHandle`] that allows us to safely pass it across threads.
 ///
 /// Depending on the platform, the underlying pointer-containing handle cannot be used on all threads,
-/// and so we cannot simply make it (or any type that has a safe operation to get a [`RawWindowHandle`] or [`RawDisplayHandle`])
+/// and so we cannot simply make it (or any type that has a safe operation to get a [`WindowHandle`] or [`DisplayHandle`])
 /// thread-safe.
-#[derive(Debug, Clone, Component)]
+#[derive(Clone, Component)]
 pub struct RawHandleWrapper {
-    _window: Arc<dyn Any + Send + Sync>,
-    /// Raw handle to a window.
-    pub window_handle: RawWindowHandle,
-    /// Raw handle to the display server.
-    pub display_handle: RawDisplayHandle,
+    window: Arc<dyn WindowTrait>,
+}
+
+impl fmt::Debug for RawHandleWrapper {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        f.debug_struct("RawHandleWrapper").finish_non_exhaustive()
+    }
 }
```

Comment:
Can this be automatically derived by `#[derive(Debug)]`, or does it complain that `dyn WindowTrait` does not implement `Debug`?

Reply:
Yeah, it's needed since `dyn WindowTrait` doesn't implement `Debug`. (The standard library has an implementation for `dyn Any`.) It seems simplest to just implement `Debug` for `RawHandleWrapper`.

Reply:
Ok, sounds good!